pkg/istio-agent/xds_proxy.go

```diff
@@ -186,6 +191,8 @@ func (p *XdsProxy) StreamAggregatedResources(downstream discovery.AggregatedDisc
 		}
 	}()
 
+	go p.healthChecker.PerformApplicationHealthCheck(healthEventsChan, p.stopChan)
```

Comment:
why is this running for all k8s pods as well? shouldn't this be gated based on whether the workload is running on VMs?

Reply:
@howardjohn am i missing something here? otherwise, it will increase load on istiod as well.

Reply:
No, should be set only if health check is configured. I commented the same inside the function.

Reply:
also, what happens if the vm did not configure any health check? we need to have a default heartbeat every 30s

Reply:
Or if the health check intervals are way too big. like every 30m. Then we cannot identify a stale from a not stale workload for gc

Reply:
i think config will only be provided for vms, and if no config is provided, NoOpHealthChecks are run, which do nothing